src/main/java/mate/academy/spring/security/CustomDetailsService.java

```diff
@@ -0,0 +1,29 @@
+package mate.academy.spring.security;
+
+import mate.academy.spring.model.User;
+import mate.academy.spring.service.UserService;
+import org.springframework.security.core.userdetails.UserDetails;
+import org.springframework.security.core.userdetails.UserDetailsService;
+import org.springframework.security.core.userdetails.UsernameNotFoundException;
+import org.springframework.stereotype.Service;
+
+@Service
+public class CustomDetailsService implements UserDetailsService {
+    private final UserService userService;
+
+    public CustomDetailsService(UserService userService) {
+        this.userService = userService;
+    }
+
+    @Override
+    public UserDetails loadUserByUsername(String email) throws UsernameNotFoundException {
```

Comment:
It's better to make this method throw UsernameNotFoundException when there is no user by email. (Exeption in method signature)